miso-web/src/main/webapp/scripts/project_ajax.js

```diff
@@ -35,5 +42,47 @@ var Project = (function () {
       });
       Project.setAssays(assays);
     },
+
+    setContacts: function (contacts) {
+      FormUtils.setTableData(ListTarget.contact, listConfig, contactsListId, contacts, form);
+    },
+
+    getContacts: function () {
+      return FormUtils.getTableData(contactsListId);
+    },
+
+    addContact: function (addContact) {
+      var contacts = Project.getContacts();
+      addContact["projectId"] = projectId;
+      if (
+        contacts.find(function (contact) {
+          return (
+            contact.contactId === addContact.contactId &&
+            contact.contactRoleId === addContact.contactRoleId
+          );
+        })
```

Comment:
This doesn't handle new contacts - if two are added with the same role, it'll call them duplicates even though the name and email are different. Change to use the same checks as in `removeContacts` (and move that to a function so it's not duplicated)